core/state/statedb.go

```diff
@@ -1407,3 +1355,19 @@ func copy2DSet[k comparable](set map[k]map[common.Hash][]byte) map[k]map[common.
 	}
 	return copied
 }
+
+func (s *StateDB) markDelete(addr common.Address) {
```

Comment:
Please add a docstring about when/why this method should be invoked (same for the one below)